docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,103 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
 **s2n_config_add_ticket_crypto_key** adds session ticket key on the server side. It would be ideal to add new keys after every (encrypt_decrypt_key_lifetime_in_nanos/2) nanos because
 this will allow for gradual and linear transition of a key from encrypt-decrypt state to decrypt-only state.
 
+## TLS1.3 Pre-Shared Key (PSK) Related Calls 
+
+S2N-TLS provides support for pre-shared keys that can be established out of band and pre-shared keys that can be established in a previous connection and then used to establish a new connection via session resumption for versions >= TLS1.3. The Pre-shared key is used to mutually authenticate the peers, under the assumption that it is not shared with others. This allows the parties to skip the certificate verification steps, saving bandwidth, communication rounds, and latency. When a PSK is used, the underlying assumption is that a party receiving a message that passes verification with the PSK knows that the message was sent by a party that also owns that PSK. 
```

Comment:
* I'd suggest clearly establishing terminology first.
* PSKs by themselves don't save any round trips
* I think the last sentence is repetitive, or I'm missing what it's trying to communicate.

```suggestion
S2N-TLS provides support for pre-shared keys (PSKs). PSKs can be established out of band (external PSKs) or in a previous connection (resumption PSKs). The PSK is used to mutually authenticate the peers, under the assumption that it is not shared with others. This allows the handshake to skip the certificate verification steps, saving bandwidth and latency.
```